src/Parthenon/DependencyInjection/Modules/Payments.php

```diff
@@ -97,14 +114,33 @@ public function handleConfiguration(array $config, ContainerBuilder $container):
         $this->configureSubscriberType($config, $container);
 
         $loader = new XmlFileLoader($container, new FileLocator(__DIR__.'/../../Resources/config'));
+
         if ('stripe' === strtolower($config['payments']['provider'])) {
             $this->handlePaymentsStripe($config, $container);
             $loader->load('services/payments/stripe.xml');
+        } elseif ('transaction_cloud' === strtolower($config['payments']['provider'])) {
+            $this->handlePaymentsTransactionCloud($config, $container);
+            $loader->load('services/payments/transaction_cloud.xml');
         }
 
         $loader->load('services/payments.xml');
     }
 
+    private function handlePaymentsTransactionCloud(array $config, ContainerBuilder $containerBuilder)
+    {
+        if (empty($config['payments']['transaction_cloud'])) {
+            throw new ParameterNotSetException('Then payment.provider is transaction_cloud then payments.transaction_cloud must be provided');
+        }
+
+        $transactionCloudConfig = $config['payments']['transaction_cloud'];
+
+        $containerBuilder->setParameter('parthenon_payments_transaction_cloud_api_key', $transactionCloudConfig['api_key'] ?? '');
```

Comment:
Should throw error if api details aren't provided but configuration is set to transaction.cloud